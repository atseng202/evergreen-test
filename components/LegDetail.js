import styles from "../styles/LegDetail.module.scss";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Card from "react-bootstrap/Card";
/**
 * LegDetail Component
 * Shows the departure and destination airport leg info
 *
 * Props:
 * leg, object like:
 *  {id, departure_aiport, arrival_airport, departure_time,
 *    arrival_time, stops, airline_name, airline_id, duration_mins}
 *
 * State: None
 *
 * FlightCard -> LegDetail
 *
 * Use https://logos.skyscnr.com/images/airlines/favicon/{id}.png
 * for airline logos for each leg
 **/
function LegDetail({ leg }) {
  /**
   * Custom helper method to add leading 0 for hours and minutes
   * less than 0
   **/
  function formatZeros(time) {
    if (time < 10) {
      return `0${time}`;
    }
    return time;
  }

  /**
   * Helper method returns text regarding number of stops
   * for this leg
   */
  function getStopsText(leg) {
    if (leg.stops == 0) {
      return "Direct";
    } else if (leg.stops == 1) {
      return leg.stops + " Stop";
    } else {
      return leg.stops + " Stops";
    }
  }

  const hours = Math.floor(leg.duration_mins / 60);
  const minutes = leg.duration_mins % 60;

  // Converting datetime string to presentational time
  const departureDate = new Date(leg.departure_time);
  const arrivalDate = new Date(leg.arrival_time);

  const formattedDepartureDate =
    formatZeros(departureDate.getHours()) +
    ":" +
    formatZeros(departureDate.getMinutes());

  const formattedArrivalDate =
    formatZeros(arrivalDate.getHours()) +
    ":" +
    formatZeros(arrivalDate.getMinutes());

  return (
    <Container className={styles.LegDetail}>
      <Row className="pt-2 pl-3 pr-3 my-auto">
        <Col xs={10}>
          {/* Row with airplane icon, departure, arrow, arrival */}
          <Row className={styles.LegDetail__dateInfo}>
            <img
              className={`pr-2 my-auto ${styles.LegDetail__airlineLogo}`}
              src={`https://logos.skyscnr.com/images/airlines/favicon/${leg.airline_id}.png`}
              alt={leg.airline_id}
            />
            {/* Disable growing for the departure time and aiprot */}
            <Col className="flex-grow-0 pr-4">
              <Row className="py-1">{formattedDepartureDate + "\n"}</Row>
              <Row className="font-weight-light">{leg.departure_airport}</Row>
            </Col>
            <img
              className={styles.LegDetail__arrowRight}
              src="/long-arrow-right.svg"
              alt="arrow-right"
            />
            <Col className="flex-grow-0 pl-4">
              <Row className="py-1">{formattedArrivalDate + "\n"}</Row>
              <Row className="font-weight-light">{leg.arrival_airport}</Row>
            </Col>
          </Row>
        </Col>
        <Col>
          <Row className="justify-content-end py-2 font-weight-light">{`${hours}h ${minutes}`}</Row>
          <Row
            className={`justify-content-end ${
              leg.stops > 0
                ? styles.LegDetail__oneOrMoreStops
                : styles.LegDetail__noStops
            }`}
          >
            {getStopsText(leg)}
          </Row>
        </Col>
      </Row>
    </Container>
  );
}

export default LegDetail;
