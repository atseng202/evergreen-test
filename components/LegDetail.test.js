import React from "react";
import { render } from "@testing-library/react";
import LegDetail from "./LegDetail";

describe("LegDetail smoke and snapshot tests", () => {
  let expectedProps;

  beforeEach(() => {
    expectedProps = {
      leg: {
        id: "leg_1",
        departure_airport: "BUD",
        arrival_airport: "LTN",
        departure_time: "2020-10-31T15:35",
        arrival_time: "2020-10-31T17:00",
        stops: 0,
        airline_name: "Wizz Air",
        airline_id: "WZ",
        duration_mins: 145,
      },
    };
  });

  it("should render correctly", () => {
    render(<LegDetail {...expectedProps} />);
  });

  it("matches LegDetail snapshot", () => {
    const { asFragment } = render(<LegDetail {...expectedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
