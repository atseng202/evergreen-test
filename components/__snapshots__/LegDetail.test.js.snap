// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegDetail smoke and snapshot tests matches LegDetail snapshot 1`] = `
<DocumentFragment>
  <div>
    <img
      src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
    />
    <div>
      15:35

      <br />
      BUD
    </div>
    <br />
    <div>
      17:00

      <br />
      LTN
    </div>
    <br />
    <div>
      2h 25
      <br />
      Direct
    </div>
  </div>
</DocumentFragment>
`;
